### PR TITLE
fix/ECMS-7935 : Fetch activity information in the document preview init

### PR DESCRIPTION
**Pull request by itouir**
Add two functions to fetch activity information in the document preview init

### CODE REVIEW

File: extension/webapp/src/main/webapp/javascript/document-preview.js [hunk 1]
```diff
@@ -70,6 +70,19 @@
...
+        // if we don't know if an activity is liked or not, let's fetch it
+        if(this.settings.activity.id != null && this.settings.activity.liked== null) {
+          promises.push(this.fetchLiked());
+        }
+        // if we don't have the activity postTime, let's fetch it
+        if(this.settings.activity.id != null && this.settings.activity.postTime == null) {
+          promises.push(this.fetchActivity());
+        }
+          // if we don't have the activity status, let's fetch it
+          if(this.settings.activity.id != null && this.settings.activity.status == null) {
```
Comment (ahamdi) [inline extension/webapp/src/main/webapp/javascript/document-preview.js]: && this.settings.activity.status == null could be added to the previous check, this check could be removed

File: extension/webapp/src/main/webapp/javascript/document-preview.js [hunk 2]
```diff
@@ -203,6 +216,36 @@
...
+        if (data.likes != null) {
+          self.settings.activity.liked = true;
+        }
+        self.clearErrorMessage();
+      }).fail(function () {
+        self.settings.activity.liked = false;
+      });
+    },
+
+      fetchActivity: function() {
```
Comment (ahamdi) [inline extension/webapp/src/main/webapp/javascript/document-preview.js]: fix indentation of this function

File: extension/webapp/src/main/webapp/javascript/document-preview.js (hunk 1 above)
Comment (itouir) [reply extension/webapp/src/main/webapp/javascript/document-preview.js]: done

File: extension/webapp/src/main/webapp/javascript/document-preview.js (hunk 2 above)
Comment (itouir) [reply extension/webapp/src/main/webapp/javascript/document-preview.js]: done

File: extension/webapp/src/main/webapp/javascript/document-preview.js [hunk 3]
```diff
@@ -70,6 +70,14 @@
         if(this.settings.activity.id != null && this.settings.activity.likes == null) {
           promises.push(this.fetchLikes());
         }
+        // if we don't know if an activity is liked or not, let's fetch it
+        if(this.settings.activity.id != null && this.settings.activity.liked== null) {
```
Comment (ahamdi) [inline extension/webapp/src/main/webapp/javascript/document-preview.js]:
```suggestion
        if(this.settings.activity.id != null && this.settings.activity.liked == null) {
```

File: extension/webapp/src/main/webapp/javascript/document-preview.js [hunk 4]
```diff
@@ -70,6 +70,14 @@
...
+          promises.push(this.fetchLiked());
+        }
+        // if we don't have the activity information, let's fetch it
+        if(this.settings.activity.id != null && (this.settings.activity.postTime == null||this.settings.activity.status == null)) {
```
Comment (ahamdi) [inline extension/webapp/src/main/webapp/javascript/document-preview.js]:
```suggestion
        if(this.settings.activity.id != null && (this.settings.activity.postTime == null || this.settings.activity.status == null)) {
```

**Comment by itouir** [comment]
This PR will applied on [PR](https://github.com/exodev/ecms/pull/516)

**Pull request closed without merging**